### PR TITLE
fix(Tree): Match design spec to show/hide selectable indicator

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

## BREAKING CHANGES

In master when we introduced the `selectable` feat for `Tree`, we were only showing indicator if it's expanded. #12362
Lately, as a fix for #14720 we merged #14916 

None of this behaviors match the design spec that says:

* Indicator is shown on hover and hidden on blur
* Selectable parent always show indicator when it's expanded. 

This PR introduces the behavior described above

ANcx1OuK15

This PR also makes all items ( parent and child ) selectable by default in a selectable `Tree`. 

#### Focus areas to test

Indicator should alway appear on Mouse Hover and Focus and hide on Mouse Leave and Blur

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams/components/Tree/treeTitleStyles.ts [hunk 1]
```diff
@@ -1,34 +1,59 @@
...
+        ...(p.selectable && {
+          [`> .${treeTitleSlotClassNames.indicator}`]: {
+            visibility: 'visible',
+          },
+        }),
+        ...borderFocusStyles[':focus'],
+      },
+      ':focus-visible': {
+        ...borderFocusStyles[':focus-visible'],
+      },
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Tree/treeTitleStyles.ts:37]:
```suggestion
      ':focus-visible': borderFocusStyles[':focus-visible'],
```

**Comment by assuncaocharles** [comment]
@pompomon @layershifter Checked all screen changes, seems correct for me.

**Comment by assuncaocharles** [comment]
@layershifter @pompomon @miroslavstastny please, check this one

**Comment by layershifter** [comment]
> In master when we introduced the `selectable` feat for `Tree`,

Was this released in `0.51.0`? If no, it's not a breaking change for customers.

File: packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx [hunk 2]
```diff
@@ -131,15 +133,18 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
...
     selectable,
     indeterminate,
     id,
   } = props;
 
+  const selectableParent = hasSubtree && selectable;
+
   const hasSubtreeItem = hasSubtree(props);
 
+  // const [showIndicator, setShowIndicator] = React.useState(false);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:146]: Cruft?

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:146]: Yes, removed

Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:146]:
```suggestion
```

Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx]:
```suggestion
```

File: packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx [hunk 3]
```diff
@@ -131,15 +133,17 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
...
     selectable,
     indeterminate,
     id,
   } = props;
 
+  const selectableParent = hasSubtree && selectable;
+
   const hasSubtreeItem = hasSubtree(props);
 
+
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:146]:
```suggestion
```

File: packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx [hunk 4]
```diff
@@ -131,13 +133,14 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
...
     selectionIndicator,
-    selectableParent,
     selected,
     selectable,
     indeterminate,
     id,
   } = props;
 
+  const selectableParent = hasSubtree && selectable;
+
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx:143]:
```suggestion
```

**Comment by assuncaocharles** [comment]
> > In master when we introduced the `selectable` feat for `Tree`,
> 
> Was this released in `0.51.0`? If no, it's not a breaking change for customers.

@layershifter We are breaking what we introduced in 0.49 in this PR #12362

**Review by layershifter — APPROVED** [review]

**Review by Ridvanovskyy — APPROVED** [review]

**Pull request merged**